journalpump/journalpump.py

```diff
@@ -675,6 +687,24 @@ def _truncate_long_message(self, json_entry):
         }
         return json.dumps(entry, default=default_json_serialization).encode("utf8")
 
+    def _apply_secret_filters(self, data):
+        changed = False
+        for idx, _ in enumerate(self.reader.secret_filters):
+            msg = data["MESSAGE"]
+            m = self.reader.secret_filters[idx]["compiled_pattern"].match(msg)
+            if m:
+                msg = ""
+                for idy, g in enumerate(m.groups()):
+                    if idy in self.reader.secret_filters[idx]["redact_groups"]:
+                        msg += self.reader.secret_filters[idx]["redaction_message"]
```

Comment:
Same here with dict access.

Reply:
resolved with a config validation step + removed idx in cases where it's not really needed ( read only loops )